src/psyclone/psyir/nodes/loop.py

```diff
@@ -576,5 +577,8 @@ def gen_code(self, parent):
         # - Add the kernel module import statements
         for kernel in self.walk(CodedKern):
             if not kernel.module_inline:
-                parent.add(UseGen(parent, name=kernel._module_name, only=True,
-                                  funcnames=[kernel._name]))
+                parent.add(UseGen(parent, name=kernel.module_name, only=True,
+                                  funcnames=[kernel.name]))
+
+        for psydata_node in self.walk(PSyDataNode):
```

Comment:
I think this fix should be in `LFRicLoop.gen_code()`. It's LFRic-specific so belongs there. (Presumably GOcean doesn't have this bug but it would be worth adding a test for this.)

Reply:
OK, but I have only moved the code that I have added, not the already existing, LFRic-specific code (I believe) that's just before that section.

Reply:
I've also added a test with gocean, and used a regex (in both tests) to actually make sure the psydata calls are in the expected place ;)

Reply:
Whoop!